File: Producer/Producer.java
package Producer;

import java.util.ArrayList;
import java.util.Random;

public class Producer {
    private int positionX;
    private int positionY;
    private int speedX;
    private int speedY;

    private void GetPosition(){
        Random rand = new Random();
        positionX = rand.nextInt(100);
        positionY = rand.nextInt(100);
    }

    private void GetSpeed(){
        Random rand = new Random();
        speedX = rand.nextInt(100);
        speedY = rand.nextInt(100);
    }
    private ArrayList<Integer> GetData(){
        ArrayList<Integer> data = new ArrayList<>();
        GetPosition();
        GetSpeed();
        data.add(positionX);
        data.add(positionY);
        data.add(speedX);
        data.add(speedY);
        return data;
    }
}

File: Producer/Producer.py
import random
import pika
import json

class Producer:
    def __init__(self):
        self.positionX = 0
        self.positionY = 0
        self.speedX = 0
        self.speedY = 0

    def get_position(self):
        self.positionX = random.randint(0, 99)
        self.positionY = random.randint(0, 99)

    def get_speed(self):
        self.speedX = random.randint(0, 99)
        self.speedY = random.randint(0, 99)

    def get_data(self):
        self.get_position()
        self.get_speed()
        data = [self.positionX, self.positionY, self.speedX, self.speedY]
        return data


    def main():
        connection = pika.BlockingConnection(pika.ConnectionParameters("localhost"))
        channel = connection.channel()
        
        queue_name = 'queue'
        channel.queue_declare(queue=queue_name)
        
        producer = Producer()
        
        data = producer.get_data()
        message = json.dumps(data)
        channel.basic_publish(exchange='',
                                routing_key=queue_name,
                                body=message)
        print(f" [x] Sent {message}")
        
        connection.close()
        
if __name__ == "__main__":
    Producer.main()
        
    
    